google/cloud/bigtable/data/_async/client.py

```diff
@@ -943,24 +943,19 @@ async def mutate_row(
                  GoogleAPIError exceptions from any retries that failed
              - GoogleAPIError: raised on non-idempotent operations that cannot be
                  safely retried.
+            - ValueError if invalid arguments are provided
         """
         operation_timeout = operation_timeout or self.default_operation_timeout
         attempt_timeout = (
             attempt_timeout or self.default_attempt_timeout or operation_timeout
         )
         _validate_timeouts(operation_timeout, attempt_timeout)
 
-        if isinstance(row_key, str):
-            row_key = row_key.encode("utf-8")
-        request = {"table_name": self.table_name, "row_key": row_key}
-        if self.app_profile_id:
-            request["app_profile_id"] = self.app_profile_id
+        if not mutations:
+            raise ValueError("No mutations provided")
+        mutations_list = mutations if isinstance(mutations, list) else [mutations]
 
-        if isinstance(mutations, Mutation):
-            mutations = [mutations]
-        request["mutations"] = [mutation._to_dict() for mutation in mutations]
-
-        if all(mutation.is_idempotent() for mutation in mutations):
+        if all(mutation.is_idempotent() for mutation in mutations_list):
```

Comment:
please leave a todo to change this to any so that we can retry the things that are retryable

Reply:
I thought mutate_row was meant to be treated as a single atomic unit. Wouldn't retrying individual mutations mess up the ordering?

Reply:
my mistake